pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -167,7 +168,7 @@ func BuildListenerTLSContext(serverTLSSettings *networking.ServerTLSSettings,
 			TLSServerRootCert:  serverTLSSettings.CaCertificates,
 		}
 
-		authnmodel.ApplyToCommonTLSContext(ctx.CommonTlsContext, certProxy, serverTLSSettings.SubjectAltNames, []string{}, ctx.RequireClientCertificate.Value)
+		authnmodel.ApplyToCommonTLSContext(ctx.CommonTlsContext, certProxy, serverTLSSettings.SubjectAltNames, []string{}, validateClient)
```

Comment:
I donot see how this is different from mutual tls mode, since for both optional tls and mtls mode, validateClient is true here

Reply:
The only difference is `RequireClientCertificate`. In Mutual TLS it sets it to true and it forces client to ask for certificate during SSL handshake.

Reply:
IC